common/network-common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -109,6 +128,27 @@ public static void deleteRecursively(File file) throws IOException {
     }
   }
 
+  private static void deleteRecursivelyUsingUnixNative(File file) throws IOException {
+    final ProcessBuilder builder = new ProcessBuilder("rm", "-rf", file.getAbsolutePath());
+    Process process = null;
+    int exitCode = -1;
+
+    try {
+      process = builder.start();
+      exitCode = process.waitFor();
```

Comment:
When you create a process, you should always consume the input and error streams. Otherwise, when the underlying buffer is full, `waitFor` will be blocked forever. See Hadoop's Shell codes for example: https://github.com/apache/hadoop/blob/trunk/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java#L905